File: leetcode/2176. Count Equal and Divisible Pairs in an Array/main.go
package main

func countPairs(nums []int, k int) int {
	var result int

	for i := range nums {
		for j := i + 1; j < len(nums); j++ {
			if (nums[i] == nums[j]) && (i*j)%k == 0 {
				result++
			}
		}
	}

	return result
}

File: leetcode/2176. Count Equal and Divisible Pairs in an Array/main.rs
impl Solution {
    pub fn count_pairs(nums: Vec<i32>, k: i32) -> i32 {
        let mut result = 0;

        for i in 0..nums.len() {
            for j in (i + 1)..nums.len() {
                if (nums[i] == nums[j]) && ((i * j) as i32) % k == 0 {
                    result += 1;
                }
            }
        }

        result
    }
}

